core/runtime/src/main/java/io/quarkiverse/operatorsdk/runtime/RunTimeOperatorConfiguration.java

```diff
@@ -59,7 +61,10 @@ public class RunTimeOperatorConfiguration {
      * specified by the kube config file the operator uses.
      * </p>
      */
-    @ConfigItem
+    // We use a default here so that we are able to detect if the configuration value is set to "". Setting the value to "" will
+    // reset the configuration and result in an empty Optional, but not setting the value at all will result in the default being
+    // applied.
+    @ConfigItem(defaultValue = QOSDK_USE_BUILDTIME_NAMESPACES)
```

Comment:
I'm wondering if it might not just be enough to set the default value to `io.javaoperatorsdk.operator.api.reconciler.Constants.WATCH_ALL_NAMESPACES` here. Ideally, we'd need a test to cover this, actually.

Reply:
If we set the default to WATCH_ALL_NAMESPACES we couldn't tell if there was no specific runtime setting, which means we would _always_ override the build time property.

Maybe the build time property doesn't matter because it's only really intended to affect the manifests and not the runtime behaviour... but at the moment the code will honour the build time property if the runtime property is not set at all.

Reply:
> Maybe the build time property doesn't matter because it's only really intended to affect the manifests and not the runtime behaviour... but at the moment the code will honour the build time property if the runtime property is not set at all.

I would say that this is the intended behavior but I'm open to change my mind about this depending on what people think. The previous version used the same property for both build and runtimes, which was conducive to confusing which version was used ultimately. With the new `generate-with-watched-namespaces` property, it is now more explicit that the build time value is used mostly for generation purposes. Its value can indeed be propagated at runtime and it's indeed how things are currently implemented, just not sure if it actually make sense since it would make it more difficult to know exactly which configuration is used at runtime. Completely separating both might make more sense with that respect.

Reply:
That was my thinking with a property that was explicitly namespaced with `manifests` in the name to separate it from properties that affect the built image.

I feel like the manifests are really runtime configuration and tbh, I think using quarkus.operator-sdk.namespaces at build time should affect the manifests and not the image, and that would make conceptual sense. It breaks the duplicated properties rule though.